### PR TITLE
HTML not sanitized in the admin home page #6492

### LINKED ISSUE #6492: HTML not sanitized in the admin home page
There are some parts of `adminHome.js` and the associated AJAX handler `AdminInstructorAccountAddAction.java` that construct HTML strings for insertion into the DOM, and concatenate user input into it without sanitizing.

The result is things like this occurring:
teammates_script_injection_bold_attempt_and_success
teammates_script_injection_alert_success

This seems more like a cosmetic issue than an actual security issue, though, since the admin using this page is assumed to be trusted.

### PR DESCRIPTION
**Pull request by thenaesh**
Fixes #6492 

Not ready for review as I'm looking to see if there are any similar issues nearby. The most obvious issues are fixed, though:
teammates_script_injection_fixed

### CODE REVIEW

**Comment by thenaesh** [comment]
Ready for review.

I know the names in the test case are extremely unrealistic (names don't have `<script>` tags in them), but the purpose of the test case is to make sure that encoding happens properly in all the fields, so I can't think of how to do this better.

In any case, I made that test case isolated from the rest, so the weird names only appear in one place.

**Comment by damithc** [comment]
> I know the names in the test case are extremely unrealistic (names don't have <script> tags in them), but the purpose of the test case is to make sure that encoding happens properly in all the fields, so I can't think of how to do this better.

@thenaesh One think you can do is to make it obvious that the data are messed up intentionally. e.g. `MaliciousName <script>` so that anybody looking at the test output won't wonder "how did that `<script>` tag show up in the name? is that a bug?"

**Comment by thenaesh** [comment]
@damithc I've changed the names in that test to reflect their malicious nature. Hopefully it's sufficient to make it obvious to anyone looking at the test output.

```
String shortNameDangerous = "<b>MaliciousInstrúctör</b>";
dangerousInstructor.name = "Malicious <script>alert('dangerous');</script>Instrúctör";
dangerousInstructor.email = "malicious.instr1<>!@gmail.tmt";
String dangerousInstitute = "TEAMMATES Malicious Institute <!@!@!>";
String dangerousDemoCourseId = "malicious.instr1___.gma-demo";
```

**Comment by junhaoyap** [comment]
@wkurniawan07 I understand peer review is ongoing, who do you think should handle reviewing this?

**Comment by damithc** [comment]
> I understand peer review is ongoing, who do you think should handle reviewing this?

@thenaesh can assign a peer or ask for volunteers in slack.

**Comment by thenaesh** [comment]
@whipermr5 Can I summon you to review this? :)

File: src/test/java/teammates/test/cases/browsertests/AdminHomePageUiTest.java [hunk 1]
```diff
@@ -101,6 +101,34 @@ private void testCreateInstructorAction() throws Exception {
...
+        String dangerousInstitute = "TEAMMATES Malicious Institute <!@!@!>";
+        String dangerousDemoCourseId = "malicious.instr1___.gma-demo";
+        
+        BackDoor.deleteAccount(TestProperties.TEST_INSTRUCTOR_ACCOUNT);
+        BackDoor.deleteCourse(dangerousDemoCourseId);
+        BackDoor.deleteInstructor(dangerousDemoCourseId, dangerousInstructor.email);
+        
+        homePage.createInstructor(shortNameDangerous, dangerousInstructor, dangerousInstitute);
+        
+        encryptedKey = BackDoor.getEncryptedKeyForInstructor(dangerousDemoCourseId, dangerousInstructor.email);
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/AdminHomePageUiTest.java]: I don't think this block is needed as we're not checking the expected join URL in this test

File: src/test/java/teammates/test/cases/browsertests/AdminHomePageUiTest.java [hunk 2]
```diff
@@ -101,6 +101,34 @@ private void testCreateInstructorAction() throws Exception {
...
+        
+        InstructorAttributes dangerousInstructor = new InstructorAttributes();
+        
+        String shortNameDangerous = "<b>MaliciousInstrúctör</b>";
+        dangerousInstructor.name = "Malicious <script>alert('dangerous');</script>Instrúctör";
+        dangerousInstructor.email = "malicious.instr1<>!@gmail.tmt";
+        String dangerousInstitute = "TEAMMATES Malicious Institute <!@!@!>";
+        String dangerousDemoCourseId = "malicious.instr1___.gma-demo";
+        
+        BackDoor.deleteAccount(TestProperties.TEST_INSTRUCTOR_ACCOUNT);
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/AdminHomePageUiTest.java]: It'll be good to repeat this block at the end of the test to clean up the created course and instructor. If you look at the last 3 lines of code, `demoCourseId` and `instructor` are deleted and that was sufficient as all the tests used this same set of IDs. However, now there is this malicious set (`dangerousDemoCourseId` and `dangerousInstructor `) that needs to be cleaned up too.

**Comment by chowyb** [comment]
@thenaesh I don't see any issues with the current changeset, but I'd just like to check if you're planning to add any more changes. If not, then I'll approve the changes here.

**Comment by thenaesh** [comment]
@chowyb No more changes planned.

**Comment by wkurniawan07** [comment]
@thenaesh as it so happens, there are some minor changes you need to do due to #6523.

**Comment by thenaesh** [comment]
@wkurniawan07 Ok, I renamed all the `Sanitizer` references to `SanitizationHelper`.

**Pull request merged**